### DISCUSSION TITLE
How to update Joomla (We can't find a download URL/Max Upload Size/Execution Time)

### DISCUSSION THREAD

**Discussion by PhocaCz** [joomla/joomla-cms#39671] [Q&A]
Hi, I am on server where Joomla 4.0.5 is installed. I want to update it to Joomla 4.2.6

1) When going to Joomla Update page, I get following message: **We can't find a download URL**

![image](https://i.imgur.com/a5XCmPx.png)

2) When going to manual update, the maximum upload size (18MB) is lower than update package (28MB). This is not my server and I have no option to change it (maybe because of security reasons).

The questions:

1) Is there no option like by extensions: Install from folder?

2) Why is the update package so large?  Is there some alternative to update?

(On this server I have problems with max upload size but on another servers I get problems when standard updating because of the execution time, etc.)

Thank you, Jan

**Comment by Fedik** [comment]
> Is there no option like by extensions: Install from folder?

Upload latest `Joomla_4.x.x-Stable-Update_Package.zip` packge in to joomla `/tmp` folder, and try again.
That should do the trick.

> Why is the update package so large? Is there some alternative to update?

Patch update not possible currently. Cannot distinguish changes in `/media` folder, so have to offer full update,

**Comment by PhocaCz** [comment]
> Upload latest Joomla_4.x.x-Stable-Update_Package.zip packge in to joomla /tmp folder, and try again.
That should do the trick.

I have uploaded this package to tmp folder but what steps should be done next. Nothing changed in Joomla Update area

**Reply by Fedik** [reply]
Have to click update as usual. The script should pick the file instead of downloading it.

**Reply by Fedik** [reply]
> We can't find a download URL

Hm, but if the script cannot check for update, then it also will not help.

**Comment by PhocaCz** [comment]
:-( I don't understand it, extensions which are usually 1 - 2 MB large have the option to install them from folder, update package which is 28MB does not have such option?

**Comment by dgrammatiko** [comment]
Go to `administrator/index.php?option=com_installer&view=updatesites` and click on the rebuild button on the toolbar. Also would helped if you provided the error Ajax response...

**Comment by PhocaCz** [comment]
I did fresh install so this was solved by reinstall. Other instance 4.1.5 the same problem.

> administrator/index.php?option=com_installer&view=updatesites Rebuild

![image](https://i.imgur.com/WHWASH5.png)

Done, nothing changed here:

![image](https://i.imgur.com/cjZF9Wn.png)

There is no AJAX when going to Components - Joomla! Update

![image](https://i.imgur.com/8vDQ66o.png)

**Reply by dgrammatiko** [reply]
> There is no AJAX when going to Components - Joomla! Update

I meant the ajax when you press the Update button...

**Comment by PhocaCz** [comment]
Which "Update" button do you mean?

**Reply by dgrammatiko** [reply]
on the Route `administrator/index.php?option=com_joomlaupdate` the `Check For Updates` button is doing some ajax calls, post the request and response to get an idea what is failing

**Comment by PhocaCz** [comment]
No AJAX, it just did reload:

![image](https://i.imgur.com/EN2nlI8.gif)

**Comment by Fedik** [comment]
That error probably means that Hosting blocked all incoming connections.

Only way I can think, is to upload files manualy via FTP, 
then remove administrator/cache/autoload_psr4.php and run "Database Fix" in backend. 

Of course, make sure you made backup before doing it.

**Comment by richard67** [comment]
@PhocaCz What's the version of your database server and which kind (MySQL or MariaDB)? Maybe it doesn't meet the minimum requirements, MySQL >= 5.6 or MariaDB >= 10.1? It is possible to make new installations of J4 with such database versions, and it works somehow later, but after that you can't update. There is a pull request to improve the handling of such cases: #38799 .

**Comment by PhocaCz** [comment]
Hi, thank you to all, I did reinstall to 4.2.6 on all instances, the message is now OK, so I will wait to test it when 4.2.7 will be ready.

**Reply by PhocaCz** [reply]
Yes, unfortunately if there is no fix, there must be workaround. I hope this will be OK in 4.2.6. Maybe this was solved in some higher version than I used (4.0.5, 4.1.5). 

But I think, the main problem here is, that there is no alternative like it is for extensions. In extension manager, you can install extension from folder (from my experience, it solves 99% of all possible problems like limits, security measures, etc.). The size of the extension is usually around 1 - 2 MB, but the size of Joomla update packages is 28MB which implies that everything will definitely not be ideal when uploading such a large file.
